### PR TITLE
Add tests that validates all current-hash.json Merkle tree files

### PR DESCRIPTION
**Pull request by mcoetzee**
Closes #7 

### What does this change?
- installs and sets up `jest` with `ts-jest`
- adds tests that validates all `current-hash.json` Merkle tree files
- includes these tests in the CI
- updates the `deriveTreeHash()` function to align with the contract tests. E.g. [/test/DapiFallbackV2.sol.js#L276](https://github.com/api3dao/dapi-management/blob/main/test/DapiFallbackV2.sol.js#L276)
- removes setting of the hash in both the sign script and endpoint following [discussion](https://api3workspace.slack.com/archives/C061WEUEXRP/p1700666961654269?thread_ts=1700659715.563089&cid=C061WEUEXRP)
- fixes the timestamp in the `sync-dapi-fallback-values.js` script
- adds signature validation in the sign script (to align with the endpoint)
- copies the `current-hash.json` to the `previous-hash.json` for the dAPI fallback Merkle tree and clears out the current signatures (because they are no longer valid)

### CODE REVIEW

**Comment by mcoetzee** [comment]
@acenolaza FYI I was reading a bit through your [contract tests](https://github.com/api3dao/dapi-management/blob/main/test/DapiFallbackV2.sol.js#L276), and saw that the frontend had a bit of a mismatch with your tests around deriving the tree hash (with the hash type, tree root, and timestamp). I've updated the frontend code to match your approach: https://github.com/api3dao/dapi-management/pull/62/files#diff-5cd225518dff9f589fb7f266964244f1b788a70e75c41bc90c6a0e01ccb56dafR68

Does this make sense to you?

**Comment by acenolaza** [comment]
> @acenolaza FYI I was reading a bit through your [contract tests](https://github.com/api3dao/dapi-management/blob/main/test/DapiFallbackV2.sol.js#L276), and saw that the frontend had a bit of a mismatch with your tests around deriving the tree hash (with the hash type, tree root, and timestamp). I've updated the frontend code to match your approach: https://github.com/api3dao/dapi-management/pull/62/files#diff-5cd225518dff9f589fb7f266964244f1b788a70e75c41bc90c6a0e01ccb56dafR68
> 
> Does this make sense to you? Feel free to leave comments 👍

First, I need to apologize for not reviewing and being up-to-date with frontend changes. Now that contracts are more or less finalized, I'll try to get up to speed with it.
Second, don't assume my work is perfect. I do make mistakes and more often that you might think 😅

File: scripts/sync-dapi-fallback-values.js [hunk 1]
```diff
@@ -23,7 +23,7 @@ async function syncDapiFallbackValues() {
     fs.writeFileSync(previousHashPath, JSON.stringify(currentHashData, null, 4));
 
     const merkleTeeData = {
-      timestamp: new Date().getTime(),
+      timestamp: Math.floor(new Date().getTime() / 1000),
```
Comment (acenolaza) [inline scripts/sync-dapi-fallback-values.js:26]: 🥇

**Review by pieterdub — APPROVED** [review]
LGTM 🚀

**Pull request merged**